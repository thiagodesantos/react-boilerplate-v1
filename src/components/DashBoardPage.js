import React from 'react';

const DashBoardPage = () => (
    <div>
        Pagina do dashboard
    </div>
);

export default DashBoardPage;