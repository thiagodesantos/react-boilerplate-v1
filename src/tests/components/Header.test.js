import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';

test('Deveria renderizar o header corretamente', () => {
    const wrapper = shallow( <Header startLogout={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Deveria chamar o startLogout no click do botÃ£o', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(<Header startLogout={startLogout}/>);
    wrapper.find('button').simulate('click');
    expect(startLogout).toHaveBeenCalled();
});

