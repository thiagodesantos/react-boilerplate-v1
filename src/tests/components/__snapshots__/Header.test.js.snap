// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deveria renderizar o header corretamente 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        to="/dashboard"
      >
        <h1>
          BoilerPlate
        </h1>
      </Link>
      <button
        className="button button--link"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
